qiskit_machine_learning/kernels/quantum_kernel.py

```diff
@@ -355,22 +357,27 @@ def construct_circuit(
                 - x and/or y have incompatible dimension with feature map
                 - unbound training parameters in the feature map circuit
         """
-        # Ensure all training parameters have been bound in the feature map circuit.
-        unbound_params = self.get_unbound_training_parameters()
-        if unbound_params:
-            raise ValueError(
-                f"""
-                The feature map circuit contains unbound training parameters ({unbound_params}).
-                All training parameters must be bound to numerical values before constructing
-                inner product circuit.
-                """
-            )
+        qc = self._construct_circuit_with_feature_map(x, y)
 
-        if len(x) != self._feature_map.num_parameters:
-            raise ValueError(
-                "x and class feature map incompatible dimensions.\n"
-                f"x has {len(x)} dimensions, but feature map has {self._feature_map.num_parameters}."
-            )
+        if y is None:
+            y = x
```

Comment:
```suggestion
        y = x if y is None
```
For compactness.

Reply:
`else` is required :)
could be `y = x if y is None else y` but I think it is better.

Reply:
You are right, my bad.